sphericart/src/sphericart_cuda.cpp

```diff
@@ -0,0 +1,188 @@
+#include <iostream>
+#include <stdexcept>
+
+#define _SPHERICART_INTERNAL_IMPLEMENTATION
+
+#include "dynamic_cuda.hpp"
+#include "sphericart.hpp"
+#include "cuda_base.hpp"
+#include "sphericart_cuda.hpp"
+
+using namespace sphericart::cuda;
+
+/*
+    This code checks whether the cuda libraries we'd need to dynamically load are available on the host
+*/
+
+void checkCuda() {
+    if (!cudadriver.loaded()) {
```

Comment:
with the current version of the code, I don't see how this will ever be `false`

Reply:
this should be fixed now with the changes above. It doesn't return false, but will throw errors if the libs aren't found.